shared/src/summarize_result.js

```diff
@@ -49,7 +49,10 @@ async function setup() {
   });
 
   browser.runtime.onMessage.addListener(async (data) => {
-    if (data.type === 'summary_finished') {
+    const searchParams = new URLSearchParams(window.location.search);
+    const url = searchParams.get('url');
+
+    if (data.type === 'summary_finished' && data.url === url) {
```

Comment:
Basically the idea here is that you can only get one summarization window/popup per URL, and only that one will accept that summarization.